go/datas/commit.go

```diff
@@ -66,6 +68,50 @@ func CommitDescendsFrom(commit types.Struct, ancestor types.Ref, vr types.ValueR
 	return true
 }
 
+// FindCommonAncestor returns the most recent common ancestor of c1 and c2, if
+// one exists, setting ok to true. If there is no common ancestor, ok is set to false.
+func FindCommonAncestor(c1, c2 types.Struct, vr types.ValueReader) (a types.Struct, ok bool) {
```

Comment:
Are these structs because they are supposed to be Commits? Could you do `d.PanicIfFalse(IsCommitType(c1.Type())` then?

(maybe we should add an IsCommitStruct so you don't need to call .Type())